protocol/urls/urls.go

```diff
@@ -123,35 +136,84 @@ func GetGithubPreviewData(link string) (previewData LinkPreviewData, err error)
 	return previewData, nil
 }
 
-func GetLinkPreviewData(link string) (previewData LinkPreviewData, err error) {
+func GetGiphyOembed(url string) (data GiphyOembedData, err error) {
+	oembedLink := fmt.Sprintf(GiphyOembedLink, url)
+
+	jsonBytes, err := GetURLContent(oembedLink)
 
-	url, err := url.Parse(link)
 	if err != nil {
-		return previewData, fmt.Errorf("Cant't parse link %s", link)
+		return data, fmt.Errorf("Can't get bytes from Giphy oembed response at %s", oembedLink)
 	}
 
-	hostname := strings.ToLower(url.Hostname())
-	youtubeHostnames := []string{"youtube.com", "www.youtube.com", "youtu.be"}
-	for _, youtubeHostname := range youtubeHostnames {
-		if youtubeHostname == hostname {
-			return GetYoutubePreviewData(link)
-		}
+	err = json.Unmarshal(jsonBytes, &data)
+	if err != nil {
+		return data, fmt.Errorf("Can't unmarshall json")
```

Comment:
This error too, to be clear the convention is error messages not the variable names.

Reply:
Understood. I was looking at the error variable name. Do you want me to change it? The errors started with an uppercase letter in other parts of this code.

Reply:
Hey sorry for the late reply, I didnt get pinged about your question and I haven't been reading my email this week.

What I normally do is if I spot something like this in preexisting code I make changes through the package. Yes please make these changes, it doesn't look like there are many, otherwise I would say just to leave them for a dedicated clean up.

Reply:
I'll make the changes to this package.